src/validators/function.rs

```diff
@@ -471,7 +470,7 @@ pub struct ValidationInfo {
     data: Option<Py<PyDict>>,
     field_name: Option<String>,
     #[pyo3(get)]
-    mode: ValidationMode,
+    mode: InputType,
```

Comment:
I realize I just created `ValidationMode` in #598, but since we're getting rid of the `get_type()` in input we can just re-use it's enum (which was the same except with one more member, which I deleted here). The good news is this will just disappear once we merge this in to #598, so the actual diff will be negative in terms of LOC!